api/v1alpha1/provisioning_types.go

```diff
@@ -99,6 +99,14 @@ type ProvisioningSpec struct {
 	// accessible from the machine networks. User should provide two IPs on
 	// the external network that would be used for provisioning services.
 	ProvisioningNetwork ProvisioningNetwork `json:"provisioningNetwork,omitempty"`
+
+	// WatchAllNamespaces provides a way to explicitly allow use of this
+	// Provisioning configuration across all Namespaces. It is an
+	// optional configuration which defaults to false and in that state
+	// will be used to provision baremetal hosts in only the
+	// openshift-machine-api namespace. When set to true, this provisioning
+	// configuration would be used for baremetal hosts across all namespaces.
+	WatchAllNamespaces bool `json:"watchAllNamespaces,omitempty"`
```

Comment:
We generally try to avoid using `bool` fields in the API, because it makes it difficult to expand or redefine them later. Here I would suggest an enum with 2 values `Single` and `All`, using `All` as the default.